wavespectra/input/spotter.py

```diff
@@ -6,16 +6,206 @@
+
+    ds['Peak Direction (deg)'].attrs['units'] = 'degree'
+    ds['Peak Direction (deg)'].attrs['standard_name'] = 'sea_surface_wave_from_direction_at_variance_spectral_density_maximum'
+    ds['Peak Direction (deg)'].attrs['long_name'] = 'Peak direction'
+
+    ds['Peak Directional Spread (deg)'].attrs['units'] = 'degree'
+    ds['Peak Directional Spread (deg)'].attrs['standard_name'] = 'sea_surface_wave_directional_spread_at_variance_spectral_density_maximum'
+    ds['Peak Directional Spread (deg)'].attrs['long_name'] = 'Peak directional spread'
+
+    ds['Mean Direction (deg)'].attrs['units'] = 'degree'
+    ds['Mean Direction (deg)'].attrs['standard_name'] = 'sea_surface_wave_from_direction'
+    ds['Mean Direction (deg)'].attrs['long_name'] = 'Mean direction'
+
+    ds['Mean Directional Spread (deg)'].attrs['units'] = 'degree'
+    ds['Mean Directional Spread (deg)'].attrs['long_name'] = 'Mean directional spread'
+
+    ds['Latitude (deg)'].attrs['units'] = 'degree_north'
+    ds['Latitude (deg)'].attrs['standard_name'] = 'latitude'
+    ds['Latitude (deg)'].attrs['long_name'] = 'Latitude'
+
+    ds['Longitude (deg)'].attrs['units'] = 'degree_east'
+    ds['Longitude (deg)'].attrs['standard_name'] = 'longitude'
+    ds['Longitude (deg)'].attrs['long_name'] = 'Longitude'
+
+    ds['Wind Speed (m/s)'].attrs['units'] = 'm/s'
+    ds['Wind Speed (m/s)'].attrs['standard_name'] = 'wind_speed'
+    ds['Wind Speed (m/s)'].attrs['long_name'] = 'Wind speed'
+
+    ds['Wind Direction (deg)'].attrs['units'] = 'degree'
+    ds['Wind Direction (deg)'].attrs['standard_name'] = 'wind_from_direction'
+    ds['Wind Direction (deg)'].attrs['long_name'] = 'Wind direction'
+
+    ds['Surface Temperature (°C)'] = 274.15*ds['Surface Temperature (°C)']
+    ds['Surface Temperature (°C)'].attrs['units'] = 'K'
+    ds['Surface Temperature (°C)'].attrs['standard_name'] = 'sea_surface_temperature'
+    ds['Surface Temperature (°C)'].attrs['long_name'] = 'Surface temperature'
+
+    ds['Frequency'].attrs['units'] = 'Hz'
+    ds['Frequency'].attrs['standard_name'] = 'wave_frequency'
+    ds['Frequency'].attrs['long_name'] = 'Frequency'
+
+    ds['Variance density'].attrs['units'] = 'm^2/Hz'
+    ds['Variance density'].attrs['standard_name'] = 'sea_surface_wave_variance_spectral_density'
+    ds['Variance density'].attrs['long_name'] = 'Spectral density'
+
+    ds['Directional spread'].attrs['units'] = 'degree'
+    ds['Directional spread'].attrs['standard_name'] = 'sea_surface_wave_directional_spread'
+    ds['Directional spread'].attrs['long_name'] = 'Directional spreading'
+    
+    ds.attrs['Conventions'] = 'CF-1.8'
+    ds.attrs['source'] = f"Sofar Spotter buoy, {filename}"
+    ds.attrs['history'] = f"generated {datetime.datetime.now().strftime('%Y-%m-%d @ %H:%M:%S')} by {getpass.getuser()}"
+    ds.attrs['references'] = 'https://content.sofarocean.com/hubfs/Technical_Reference_Manual.pdf'
+
+
+    ds = ds.rename({'Epoch Time':'time',
```

Comment:
It looks like I should maybe set variable names and `attrs` using [`attributes.yml`](https://github.com/wavespectra/wavespectra/blob/master/wavespectra/core/attributes.yml) -- I see this is being done in the `JSON` workflow, but don't immediately understand how to do it in the workflow I'm using the for `CSV` files

Reply:
There is a function to help with attribute definition, it uses that attributes.yml as you correctly noted. You import it from here (it is already imported in spotter.py actually):

```
from wavespectra.core.attributes import set_spec_attributes
```

and then you just run something like:

```
set_spec_attributes(dset)
```

before returning dset in your reader function.

Reply:
However it looks like you also return several other non-spectral variables in the output - those won't have attributes defined in attributes.yml so you will have to manually define that for those as you are doing.

Reply:
Since I already do this manually and, as you note, include many other variables, I'm going to forgo the `set_spec_attributes(dset)` method unless that's an issue for you.